dpnp/dpnp_iface_indexing.py

```diff
@@ -81,35 +81,29 @@ def choose(x1, choices, out=None, mode='raise'):
     --------
     :obj:`take_along_axis` : Preferable if choices is an array.
     """
-    if not use_origin_backend(x1):
-        if not isinstance(x1, list):
-            pass
-        elif not isinstance(choices, list):
+    x1_desc = dpnp.get_dpnp_descriptor(x1)
+    
+    choices_list =[]
+    for choice in choices:
+        choices_list.append(dpnp.get_dpnp_descriptor(choice))
+
+    size = x1_desc.size
+    choices_size = choices_list[0].size
```

Comment:
`dpnp.get_dpnp_descriptor` can return `False` if no conversion possible. In this case, it should go to fallback